### PR TITLE
fix(agent): bind published container ports to the node's private address, not 0.0.0.0

### PR DESCRIPTION
**Pull request by dviejokfs**
## Summary

- Every container a worker node's `temps agent` published — both app deploys (via `DockerRuntime`) and services created directly through the agent's HTTP API (managed databases, Redis, etc., via `service_handlers`) — bound its host port to `0.0.0.0`, all interfaces, including any public one. That bypasses the control-plane proxy (auth, TLS termination, routing) entirely: anyone who can reach the node's public IP directly could hit the raw container port.
- `temps join` already registers a node's private/overlay address (`nodes.private_address` — the WireGuard tunnel IP in relay mode, the operator-supplied address in direct mode) and now saves it into `agent.json`. `temps agent` requires it at startup (hard error with a recovery path if a legacy `agent.json` predates the field) and binds both app and service containers to it instead of `0.0.0.0`. Direct-mode nodes that register a non-RFC1918 address get a startup warning recommending a host firewall, since that mode intentionally allows a public IP with no WireGuard underlay.
- This is the agent-side half of the fail-closed check already living in `deploy_image.rs`, which refuses to retain or route an app container unless it can confirm the worker actually bound it to the private address — every deployment to an agent node has been failing against that check until this fix ships.

## Test plan

- [x] `cargo check --lib -p temps-agent -p temps-cli -p temps-deployer` — clean
- [x] `cargo test -p temps-agent --lib` — 106 passed (includes new regression test)
- [x] `cargo test -p temps-cli --lib -- commands::agent:: commands::join::` — 6 passed
- [x] `cargo test -p temps-deployer --lib -- docker::` — 79 passed
- [x] `$(command -v cargo) clippy -p temps-agent -p temps-cli -p temps-deployer --lib --tests -- -D warnings` — clean
- [x] `cargo fmt --check` on all touched crates — clean
- [x] Security-reviewed in two passes (core `DockerRuntime`/`temps join`/`temps agent` fix, then the `service_handlers.rs` extension for service containers) — no blocking findings; added a unit test (`service_port_binding_tests::never_binds_to_all_interfaces`) guarding against the exact hardcoded-`0.0.0.0` regression this fixes

### CODE REVIEW

File: crates/temps-cli/src/commands/agent.rs [hunk 1]
```diff
@@ -68,6 +68,18 @@ pub struct AgentCommand {
...
+    /// plane during `temps join` (`nodes.private_address`) — the WireGuard
+    /// tunnel IP in relay mode, or the user-managed address in direct mode.
+    /// Published Docker container ports are bound to this address only,
+    /// never to "0.0.0.0", so deployed containers are reachable from the
+    /// control-plane proxy over the private network but never on this
+    /// node's public interface. Overrides the saved config; must match what
+    /// was registered with the control plane, since that's the address the
+    /// proxy dials for this node.
+    #[arg(long, env = "TEMPS_AGENT_PRIVATE_ADDRESS")]
+    pub private_address: Option<String>,
```
[the remark this answers is not in the document]
Comment (dviejokfs) [reply crates/temps-cli/src/commands/agent.rs:82]: Not a new pattern — this follows the exact same shape as every other field on `AgentCommand` in this file: `listen_address` (`TEMPS_AGENT_ADDRESS`), `token` (`TEMPS_AGENT_TOKEN`), `node_name` (`TEMPS_NODE_NAME`), `control_plane_url` (`TEMPS_CONTROL_PLANE_URL`), `node_id` (`TEMPS_NODE_ID`), `underlay_dev` (`TEMPS_AGENT_UNDERLAY_DEV`), `underlay_mtu` (`TEMPS_AGENT_UNDERLAY_MTU`) — all `#[arg(long, env = "...")]`, all pre-existing, none touched by this PR.

CLAUDE.md's "no new runtime configuration as environment variables" rule targets *per-tenant/per-record* behavior toggles that should be admin-editable via API/UI with audit logging (its own example: `oidc_providers.trust_idp_email`, not `TEMPS_OIDC_SKIP_EMAIL_VERIFIED`). It explicitly carves out "bootstrap-time config needed before a database connection exists" as the sanctioned exception. `temps agent` is a headless worker daemon with no database of its own — its *entire* configuration surface is exactly this kind of pre-DB bootstrap config, which is why every field on this struct already uses this pattern. `private_address` isn't a new category, it's the same category as its seven siblings.

It's also not a silently-diverging second source of truth: it's an explicit, documented override with the same precedence as every other field (CLI/env > saved `agent.json`), for the same documented purpose those siblings already exist for — deployments that don't persist `agent.json` (e.g. ephemeral/containerized agents) or explicit recovery from a stale file.

Leaving as-is; happy to revisit if there's a specific divergence scenario in mind that the other seven fields don't already share.

File: crates/temps-deployments/src/handlers/nodes.rs [hunk 2]
```diff
@@ -652,15 +666,19 @@ fn validate_node_private_address(addr: &str) -> Result<(), NodeAddressError> {
...
+        // an IPv6 host:port, so an unbracketed address is unambiguous by
+        // colon count alone: any bare IPv6 address needs at least two
+        // colons (minimum form "::"), so exactly one colon can only mean
+        // IPv4:port.
+        if addr.matches(':').count() == 1 {
+            addr.rsplit_once(':')
+                .map_or(addr, |(before, _after)| before)
         } else {
             addr
         }
```
[the remark this answers is not in the document]
Comment (dviejokfs) [reply crates/temps-deployments/src/handlers/nodes.rs:684]: Fixed in 3941c09e (route_table.rs's build_container_backend_addr and join.rs's direct-mode "address" field now use SocketAddr's Display, which brackets IPv6 automatically) and refined in 93ef7b6e (agent_listen_port/socket_authority extracted as tested helpers, fixing a related bracket-handling edge case in --agent-address parsing). Verified via new regression tests: test_build_container_backend_addr_remote_brackets_ipv6, socket_authority_brackets_ipv6, agent_listen_port_falls_back_when_bracketed_ipv6_has_no_port. Current head (93ef7b6e) scored 5/5, explicitly listing "Correctly brackets IPv6 authorities used for agent registration and container proxy routing" as verified. Resolving.
